app/models/invitation.rb

```diff
@@ -0,0 +1,6 @@
+class Invitation < ApplicationRecord
+  belongs_to :attendee, class_name: 'User'
+  belongs_to :attended_event, class_name: 'Event'
+  scope :past_events, -> { where(past_events: DateTime.now) }
+  scope :upcoming_events, -> { where(upcoming_events: DateTime.now) }
```

Comment:
Do we need these scopes here?

Reply:
Yes, the `invitation model` is working as our 'through_table or `event_attendance model`. It's an attempt to filter past and upcoming events for the attendee.